src/Features/Core/Portable/EditAndContinue/ProjectDiagnostics.cs

```diff
@@ -24,7 +24,7 @@ public static ImmutableArray<DiagnosticData> ToDiagnosticData(this ImmutableArra
             foreach (var diagnostic in projectDiagnostics)
             {
                 var document = solution.GetDocument(diagnostic.Location.SourceTree);
-                var data = (document != null) ? DiagnosticData.Create(diagnostic, document) : DiagnosticData.Create(solution, diagnostic, project);
+                var data = document != null ? DiagnosticData.Create(diagnostic, document) : DiagnosticData.Create(diagnostic, project);
```

Comment:
Can we add interface `IDiagnosticTarget` or something like that that both Document and Project implement?

Reply:
Could that simplify various conditional code?

Reply:
We have ProjectOrDocumentId.  But not some sort of ProjectOrDocument.  I'm not opposed to that, but i'd prefer to keep that for a future PR.  Thanks!